datafusion/optimizer/src/single_distinct_to_groupby.rs

```diff
@@ -478,4 +520,52 @@ mod tests {
 
         assert_optimized_plan_equal(&plan, expected)
     }
+
+    #[test]
+    fn two_distinct_and_one_common() -> Result<()> {
+        let table_scan = test_table_scan()?;
+
+        let plan = LogicalPlanBuilder::from(table_scan)
+            .aggregate(
+                vec![col("a")],
+                vec![
+                    sum(col("c")),
+                    count_distinct(col("b")),
+                    Expr::AggregateFunction(expr::AggregateFunction::new(
```

Comment:
Perhaps this could use https://docs.rs/datafusion/latest/datafusion/logical_expr/expr_fn/fn.max.html ?